internal/nginx/modules/httpmatches_test.js

```diff
@@ -0,0 +1,243 @@
+import redirect from './httpmatches.js';
+let expect = require('chai').expect;
+
+// NGINX HTTP Request Object.
+//See documentation for all properties available: http://nginx.org/en/docs/njs/reference.html
+let r;
+
+describe('redirect', function () {
+  beforeEach(function () {
+    r = {
+      method: 'GET',
+      variables: {},
+      args: {},
+      headersIn: {},
+      // properties added for testing
+      testReturned: 0,
+      testRedirectedTo: '',
+    };
+
+    // Test mocks
+    r.return = function (statusCode) {
+      r.testReturned = statusCode;
+    };
+
+    r.internalRedirect = function (redirectPath) {
+      r.testRedirectedTo = redirectPath;
+    };
+
+    r.error = function (msg) {
+      console.log('\terror:', msg);
+    };
+
+    r.warn = function (msg) {
+      console.log('\twarn:', msg);
+    };
+  });
+  it('returns 404 if http_matches variable is not defined', function () {
+    redirect.redirect(r);
+    expect(r.testReturned).to.equal(404);
+  });
+  it('returns 404 if http_matches is empty', function () {
+    r.variables = {
+      http_matches: '{}',
+    };
+    redirect.redirect(r);
+    expect(r.testReturned).to.equal(404);
+  });
+  it('redirects to the path if no conditions are defined in match', function () {
+    let matches = {
+      path: 'no-conditions',
+    };
+    r.variables = {
+      http_matches: JSON.stringify(matches),
+    };
+
+    redirect.redirect(r);
+    expect(r.testRedirectedTo).to.equal('no-conditions');
+  });
+  it('returns 405 if method does not match', function () {
+    let matches = {
+      method: 'GET',
+      path: 'get-location',
+    };
+    r.method = 'POST';
+    r.variables = {
+      http_matches: JSON.stringify(matches),
+    };
+    redirect.redirect(r);
+    expect(r.testReturned).to.equal(405);
+  });
+  it('redirects to match path if method matches', function () {
+    let matches = {
```

Comment:
in this test we have two similar test cases. I wonder if the framework can allow us to define a table and as a result, remove some duplication and make the code clearer.   Something similar to https://github.com/nginxinc/nginx-kubernetes-gateway/blob/1c650a7ac770b5a39eed46ac8aacefc6dad9a396/internal/events/loop_test.go#L263

Reply:
Good suggestion. I'll rework these a bit to look more like table-driven tests and reduce the duplication.

Reply:
I refactored all the tests to be table-style tests since the code flow was the same for all of them.